source/plugin/commands/argumentparsing.rst

```diff
@@ -0,0 +1,177 @@
+================
+Argument Parsing
+================
+
+.. javadoc-import::
+    org.spongepowered.api.command.parameter.CommandContext
+    org.spongepowered.api.command.parameter.Parameter
+    org.spongepowered.api.command.parameter.Parameter.Builder
+    org.spongepowered.api.command.Command.Builder
+    java.lang.Class
+
+Minecraft's Brigadier system includes a powerful argument parser that Sponge can take advantage of. 
+It converts the string input to java base types (integer, booleans, string) or game objects 
+(players, worlds, block types, ...). Sponge's system also supports optional arguments and flags. It also handles 
+suggestions of arguments.
+
+The parsed arguments are stored in the :javadoc:`CommandContext` object. If the parser returns a single object, 
+obtain it with :javadoc:`CommandContext#one(ResourceKey)`. Optional and weak arguments may return ``Optional.empty()``
+
+Many of the parsers may return more than one object; for example, multiple players with a matching username. In that 
+case, you must use the :javadoc:`CommandContext#all(ResourceKey)` method to get the ``Collection`` of possible matches. 
```

Comment:
I couldn't find that method. Only when:

ResourceKey -> Parameter.Key

Reply:
Changed